src/basic/in-addr-util.c

```diff
@@ -464,3 +464,40 @@ int in_addr_mask(int family, union in_addr_union *addr, unsigned char prefixlen)
 
         return -EAFNOSUPPORT;
 }
+
+int in_addr_prefix_from_string(const char *p, int *family, union in_addr_union *prefix, uint8_t *prefixlen) {
+        union in_addr_union buffer;
+        const char *e, *l;
+        uint8_t k;
+        int f, r;
+
+        assert(p);
+
+        /* prefixlen */
+        e = strchr(p, '/');
+        if (e) {
+                unsigned i;
+
+                r = safe_atou(e + 1, &i);
+                if (r < 0)
+                        return r;
+
+                if (i > 128)
```

Comment:
yes since the family is not determined yet which is done in later stage  , here looking for he maximum for IPv4 and IPv6.